src/fileListUnzip.ts

```diff
@@ -22,3 +31,18 @@ export async function fileListUnzip(fileList: FileList[]) {
 
   return fileList;
 }
+
+async function checkIfZip(file: FileList, checkZip: boolean) {
+  if (checkZip) {
+    const buffer = await file.arrayBuffer();
+    const signature = getFileSignature(buffer);
+    return { isZipped: signature === FILES_SIGNATURES.ZIP, buffer };
+  }
+  return { isZipped: /\.zip$/.exec(file.name) };
+}
+
+function getFileSignature(fileArrayBuffer: ArrayBuffer) {
+  return new Uint8Array(fileArrayBuffer)
+    .slice(0, 4)
+    .reduce((acc, byte) => (acc += byte.toString(16).padStart(2, '0')), '');
```

Comment:
wouldn't it be easier to compare numbers? and store the expected signature as an array of bytes: `[0x50, 0x4b, 0x03, 0x04]`)

Btw, according to this code, there are multiple possibilities for the magic bytes: https://github.com/kevva/is-zip/blob/master/index.js

Reply:
many thanks for it @targos

Reply:
@jobo322 But you didn't implement the approach of Michael. I will do because I would like to merge this PR.